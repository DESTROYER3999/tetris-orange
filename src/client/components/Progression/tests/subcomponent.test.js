import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ProgressionSub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<LoginSub />", () => {
  test("ProgressionSub rendering with a player in game", () => {
    const wrapper = shallow(<ProgressionSub score={45123} level={12} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
