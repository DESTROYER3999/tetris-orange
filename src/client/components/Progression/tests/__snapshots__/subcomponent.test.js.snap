// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> ProgressionSub rendering with a player in game 1`] = `
<styled.div
  id="Progression-container"
>
  <styled.div
    id="Score-container"
  >
    <div
      className="score-title"
    >
      <p>
        Score
      </p>
    </div>
    <div
      className="score-value"
    >
      <span>
        45123
      </span>
    </div>
  </styled.div>
  <styled.div
    id="Level-container"
  >
    <span>
      Level 
      12
    </span>
  </styled.div>
</styled.div>
`;
